t/app/controller/report_inspect.t

```diff
@@ -129,6 +129,21 @@ FixMyStreet::override_config {
         $report->update({ state => $old_state });
     };
 
+    subtest "changing state subscribes inspector to alerts" =>sub {
```

Comment:
The alert was actually added in the test around line 60, the first time the form is submitted with a state change. You could check that the alert is added there as an addition to that test; this test is only checking that another state change submission doesn't create another alert, not what it says it is testing. I don't know what the `$old_state`, deleting comments, or `discard_changes` are doing, this feels like it's been copied-and-pasted.